### PR TITLE
chore!: module from template and tests updated

### PR DESCRIPTION
**Pull request by waxb**
- chore: changelog updated
- chore: module from template and tests updated
- fix: `initial_group_config` validation

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 
 - Add instructions on how to use memberships to the example in the README
+- Tests added
+- README updated
+
+### Fixed
+
+- `initial_group_config` validation excluded 'INITIAL_GROUP_CONFIG_UNSPECIFIED', now it is accepted
```
Comment (mariux) [inline CHANGELOG.md:18]:
```suggestion
- Fix `initial_group_config` validation to include 'INITIAL_GROUP_CONFIG_UNSPECIFIED'
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 
 - Add instructions on how to use memberships to the example in the README
+- Tests added
+- README updated
```
Comment (mariux) [inline CHANGELOG.md:14]:
```suggestion
- Add unit tests
```

File: versions.tf [hunk 3]
```diff
@@ -1,10 +1,14 @@
+# ----------------------------------------------------------------------------------------------------------------------
+# SET TERRAFORM AND PROVIDER REQUIREMENTS FOR RUNNING THIS MODULE
+# ----------------------------------------------------------------------------------------------------------------------
+
 terraform {
-  required_version = ">= 0.14.7, < 2.0"
+  required_version = "~> 1.0"
```
Comment (mariux) [inline versions.tf:6]: change is missing in change log and this is a breaking change! ;)

File: variables.tf [hunk 4]
```diff
@@ -42,8 +42,8 @@ variable "initial_group_config" {
   default     = "EMPTY"
 
   validation {
-    condition     = contains(["WITH_INITIAL_OWNER", "EMPTY"], var.initial_group_config)
-    error_message = "The value must only be one of these valid values: WITH_INITIAL_OWNER, and EMPTY."
+    condition     = contains(["WITH_INITIAL_OWNER", "EMPTY", "INITIAL_GROUP_CONFIG_UNSPECIFIED"], var.initial_group_config)
```
Comment (mariux) [inline variables.tf:45]: nice catch

File: test/unit-minimal/_generated_google.tf [hunk 5]
```diff
@@ -36,25 +41,31 @@ provider "google" {
...
-  billing_account           = var.gcp_billing_account
-  org_directory_customer_id = try(data.google_organization.org[0].directory_customer_id, null)
-  org_domain                = var.gcp_org_domain
-  org_id                    = try(data.google_organization.org[0].name, null)
-  project_id                = var.gcp_project
-  project_number            = data.google_project.project.number
-  random_suffix             = try(random_id.suffix[0].hex, "disabled")
-  test_name                 = "unit-minimal"
+  billing_account       = var.gcp_billing_account
+  directory_customer_id = try(data.google_organization.org[0].directory_customer_id, "no-directory-customer")
```
Comment (mariux) [inline test/unit-minimal/_generated_google.tf:64]: please do not rename the variable.

**Review by mariux — CHANGES_REQUESTED** [review]

File: test/unit-minimal/_generated_google.tf (hunk 5 above)
Comment (waxb) [reply test/unit-minimal/_generated_google.tf:64]: it was `directory_customer_id` in the template both getting value from `google_organization.org[0].directory_customer_id` so this seems the consistent way. what do you think?

Comment (waxb) [reply test/unit-minimal/_generated_google.tf:64]: https://github.com/mineiros-io/terraform-google-identity-group/blob/7e3ac9d268bf1388a63d322356d341a6ce7308ce/test/terramate_google.tm.hcl#L75


https://github.com/mineiros-io/terraform-module-template/blob/b139cde8b396283a130e7df93693e2e1a8ccd335/test/terramate_google.tm.hcl#L83

Comment (waxb) [reply test/unit-minimal/_generated_google.tf:64]: also changed in the relevant parts (e.g.: https://github.com/mineiros-io/terraform-google-identity-group/blob/3eb295990195bc690274c483f0fe65cf171b6c42/test/unit-complete/main.tf#L8 )

**Review by zied-elouaer — APPROVED** [review]
LGTM!

**Review by mariux — APPROVED** [review]
lgtm

**Pull request merged**